src/Spago/Build.hs

```diff
@@ -282,3 +288,19 @@ search = do
   let cmd = "node .spago/purescript-docs-search search"
   echoDebug $ "Running `" <> cmd <> "`"
   viewShell $ callCommand $ Text.unpack cmd
+
+-- | If we are using the --sharedOutput flag, calculate the extra args to 
+-- | send to Purs compile
+getBuildArgsForSharedFolder 
+  :: Spago m 
+  => m [Purs.ExtraArg]
+getBuildArgsForSharedFolder = do
+  configPath <- asks globalConfigPath
+  outputFolder <- PackageSet.findRootOutputPath (Text.unpack configPath)
+  case pathToOutputArg <$> outputFolder of
+    Just newArg -> pure [newArg]
+    _           -> pure []
+
+-- | Take root output path and turn it into Purs argument
+pathToOutputArg :: String -> Purs.ExtraArg
+pathToOutputArg = Purs.ExtraArg . Text.pack . ((++) "-o ")
```

Comment:
Let's move it to a `let` clause inside the previous function since we use it only once